src/Service/ProductImporterManager.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\ProductCategory;
+use App\Repository\ProductCategoryRepository;
+use League\Csv\Reader;
+use App\Entity\Product;
+use App\Repository\ProductRepository;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\EntityManagerInterface;
+
+
+class ProductImporterManager
+{
+    /**
+     * @var ProductRepository
+     */
+    private $productRepository;
+
+    private $em;
+    private $msg;
+
+    public function __construct(ProductRepository $productRepository, EntityManagerInterface $entityManager)
+    {
+        $this->productRepository = $productRepository;
+        $this->em = $entityManager;
+        $this->msg = ['msq'];
+    }
+
+    public function importData($file): void
+    {
+        $csv = Reader::createFromPath($file, 'r');
+        $csv->setHeaderOffset(0);
+
+        foreach ($csv as $record){
+            if(!$this->productRepository->find($record['id'])) {
+                $product = new Product();
+                $product->setName($record['name']);
```

Comment:
you can create new method in entity which will set bunch of data at once. Something like $product->setDatafromArray($record)